frontend/app/controllers/Giraffe.scala

```diff
@@ -110,3 +118,15 @@ object Giraffe extends Controller {
     })
   }
 }
+
+object MakeURL {
```

Comment:
Have you seen the netaporter Scala Uri API? 

import com.netaporter.uri.dsl._
import com.netaporter.uri.Uri

With that you can do something like this:

var x = Uri.parse(request.uri).withScheme("https")
val url = x.copy(pathParts = Seq(PathPart(countryGroup.id)) ++ x.pathParts)